src/typescript-reporter/reporter/ControlledWatchSolutionBuilderHost.ts

```diff
@@ -0,0 +1,107 @@
+import * as ts from 'typescript';
+import fs from 'graceful-fs';
+import { createControlledWatchCompilerHost } from './ControlledWatchCompilerHost';
+import { ControlledWatchHost } from './ControlledWatchHost';
+import { TypeScriptHostExtension } from '../extension/TypeScriptExtension';
+
+interface ControlledWatchSolutionBuilderHost<TProgram extends ts.BuilderProgram>
+  extends ts.SolutionBuilderWithWatchHost<TProgram>,
+    ControlledWatchHost {}
+
+function createControlledWatchSolutionBuilderHost<TProgram extends ts.BuilderProgram>(
+  configFileName: string,
+  optionsToExtend: ts.CompilerOptions | undefined,
+  system: ts.System,
+  createProgram?: ts.CreateProgram<TProgram>,
+  reportDiagnostic?: ts.DiagnosticReporter,
+  reportWatchStatus?: ts.WatchStatusReporter,
+  reportSolutionBuilderStatus?: (diagnostic: ts.Diagnostic) => void,
+  afterProgramCreate?: (program: TProgram) => void,
+  afterProgramEmitAndDiagnostics?: (program: TProgram) => void,
+  hostExtensions: TypeScriptHostExtension[] = []
+): ControlledWatchSolutionBuilderHost<TProgram> {
+  const controlledWatchCompilerHost = createControlledWatchCompilerHost(
+    configFileName,
+    optionsToExtend,
+    system,
+    createProgram,
+    reportDiagnostic,
+    reportWatchStatus,
+    afterProgramCreate,
+    hostExtensions
+  );
+
+  let controlledWatchSolutionBuilderHost: ControlledWatchSolutionBuilderHost<TProgram> = {
+    ...controlledWatchCompilerHost,
+    writeFile(path: string, data: string, writeByteOrderMark?: boolean): void {
+      // return ts.sys.writeFile(fileName, data, writeByteOrderMark);
+    },
+    deleteFile(path: string): void {
+      // if (ts.sys.deleteFile) {
```

Comment:
Is the commented code intentional?

Reply:
Yes, it's temporary. I think I will add a configuration option to choose between "passive" and "active" mode for `build: true`. The default setting will be a "passive" mode in which the plugin doesn't write to the file system.  In this case, it will be a responsibility of the `ts-loader` or `babel-loader`

Reply:
This sounds interesting; could you tell me more about the use case for `build: true` and `build: false` please?

Reply:
`build` flag is an equivalent to the `--build` flag in `tsc` command. When `build: true`, we use `SolutionBuilder` which consists of many `BuilderProgram`s (project references) in opposition to the `build: false` where we have only one `BuilderProgram`.  In the `build: true` mode TypeScript will try to build project references if they are not built or are not up to date. In the `build: false` mode TypeScript will throw an error if project references are not built and will not check if they are up to date with their source.
The `SolutionBuilder` API is available starting from TypeScript 3.6